04week/mastermind.js

```diff
@@ -17,31 +17,56 @@ function printBoard() {
   }
 }
 
-function generateSolution() {
-  for (let i = 0; i < 4; i++) {
-    const randomIndex = getRandomInt(0, letters.length);
-    solution += letters[randomIndex];
-  }
-}
+// function generateSolution() { //creates the randomized solution 
+//   for (let i = 0; i < 4; i++) {
+//     const randomIndex = getRandomInt(0, letters.length);
+//     solution += letters[randomIndex];
+//   }
+// }
+
+// function getRandomInt(min, max) { //creates rando interger to use in generate soln?
+//   return Math.floor(Math.random() * (max - min)) + min;
+// }
 
-function getRandomInt(min, max) {
-  return Math.floor(Math.random() * (max - min)) + min;
-}
 
-function generateHint() {
-  // your code here
+function generateHint(guess) {
+  let countExact = 0; 
+  let countLetter = 0; 
+  // could i have used: for (let i of guess){} ? //
+  for (let i= 0; i<= guess.length; i++) {
```

Comment:
you probably want to do 2 separate loops since you want to make sure to give the points for an exact match before checking to see if there are any that have the correct letter at the wrong spot.